### PR TITLE
[macOS] File icons: add separate icons for gdscript, resource, scene and projects.

### PR DESCRIPTION
**Pull request by bruvzg**
Before:

old

After:
screenshot 2018-12-03 at 20 51 47

screenshot 2018-12-03 at 20 52 02

### CODE REVIEW

File: misc/dist/osx_tools.app/Contents/Info.plist [hunk 1]
```diff
@@ -90,6 +90,57 @@
...
+			<key>UTTypeIconFile</key>
+			<string>Resource.icns</string>
+			<key>UTTypeConformsTo</key>
+			<array>
+				<string>public.data</string>
+			</array>
+			<key>UTTypeTagSpecification</key>
+			<dict>
+				<key>com.apple.ostype</key>
+				<string>GDRC</string>
```
Comment (akien-mga) [inline misc/dist/osx_tools.app/Contents/Info.plist]: Does the specification require such short identifiers? I know the existing one for the `project.godot` file was `GODP`, but it seems pretty cryptic to me. If we can use longer identifiers, something like `Godot Resource` would be better IMO (or `GodotResource` if spaces are an issue).

File: misc/dist/osx_tools.app/Contents/Info.plist [hunk 2]
```diff
@@ -90,6 +90,57 @@
...
+				<string>script/x-gd</string>
+			</dict>
+		</dict>
+		<dict>
+			<key>UTTypeIdentifier</key>
+			<string>public.res</string>
+			<key>UTTypeReferenceURL</key>
+			<string></string>
+			<key>UTTypeDescription</key>
+			<string>Godot Script</string>
```
Comment (akien-mga) [inline misc/dist/osx_tools.app/Contents/Info.plist]: Should be `Godot Resource`.

File: misc/dist/osx_tools.app/Contents/Info.plist [hunk 3]
```diff
@@ -90,6 +90,57 @@
...
+			<key>UTTypeTagSpecification</key>
+			<dict>
+				<key>com.apple.ostype</key>
+				<string>GDSC</string>
+				<key>public.filename-extension</key>
+				<array>
+					<string>gd</string>
+				</array>
+				<key>public.mime-type</key>
+				<string>script/x-gd</string>
```
Comment (akien-mga) [inline misc/dist/osx_tools.app/Contents/Info.plist]: Can we be more explicit on the declare MIME type identifiers too? e.g. `script/x-gdscript`.

BTW is `script/` the correct category?
On Linux I have:
```
$ echo "/usr/bin/python" > test.py && mimetype test.py
test.py: text/x-python
```

File: misc/dist/osx_tools.app/Contents/Info.plist (hunk 1 above)
Comment (bruvzg) [reply misc/dist/osx_tools.app/Contents/Info.plist]: OSType is 4-byte, but these ID are only used for Mac OS Classic types, and can be removed at all.

File: misc/dist/osx_tools.app/Contents/Info.plist (hunk 3 above)
Comment (bruvzg) [reply misc/dist/osx_tools.app/Contents/Info.plist]: Changed to `text/x-gdscript`, `application/x-godot-resource`, `application/x-godot-scene` and `text/x-godot-project`

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**